### ISSUE TITLE
Valve Index default key bindings are inconvenient

### ISSUE THREAD

**Issue by Narazaka** [Yellow-Dog-Man/Resonite-Issues#462]
### Is your feature request related to a problem? Please describe.

Resonite requires a lot of grabbing, and the default binding of the Valve Index controller is inconvenient because it is easy to drop things.

### Describe the solution you'd like

Maybe change the threshold for grabbing?

### Describe alternatives you've considered

It might be possible to assign operations in the inspector to things that are not "grabs".

**Comment by Nytra** [comment]
You can change the controller bindings through SteamVR to make gripping easier

**Comment by Banane9** [comment]
To get around this, most Index users I know use @TehTurk's bindings - I also recommend them to people I introduce to the game.

**Comment by Frooxius** [comment]
We recommend tweaking these values for your controllers in Steam VR.

In our experience, the grip sensitivity can vary quite a bit between specific controllers - the current values were tweaked based on feedback from a number of people with Index controllers.

Lowering the threshold further can result in some controllers gripping too easily, resulting in a "sticky hands" phenomenon, where it's difficult to stop grabbing.

**Comment by Banane9** [comment]
I've only ever experienced the opposite - people having issues picking up items or dropping them on accident.

But the bindings also fix another problem with the default Index bindings. Namely that they require fully pulling the primary ("clicking it") to trigger an action. No other VR game I know of does that. It's very irritating when trying to type fast and unusual for other interactions too.

**Comment by XDelta** [comment]
IIRC we had this tweaked to not require full click in and much lower strength during durian so the default bindings weren't bad. May be something that didn't carry over? I've been using my own adjustments so not sure what the defaults are like currently.

**Comment by Frooxius** [comment]
@Banane9 Yes, that is on purpose, because we avoid setting it too low to have that issue happen now.

The main root of the issue is that the grip sensitivity is variable between Index controllers. Some require stronger, some weaker. There's two approaches:
- We set it pretty low, making it always easy to grip - but now it's too hard to not-grip on some controllers
- We set it a bit higher, making it easy not to grip - but now some controllers require stronger grip force

Neither of these is ideal, but out of the two options, the latter is a bit less disruptive, so we went for that one. The expectation is that users would tweak those settings a bit to match their controllers better.

I think best approach here would be one of the following:
- Allow to change these in Settings UI in Resonite itself - maybe even during initial setup
- Have some calibration mechanism during setup, that checks where the grip happens
- Add heuristics for the grip (but this can also make things weirder)

**Comment by JackTheFoxOtter** [comment]
I can only speak for myself, but I for once massively prefer Resonite's defaults over those of other VR games. In VRChat I frequently got frustrated because I was constantly accidentally grabbing stuff and couldn't let go of it reliably, I've never experienced any similar frustration in Resonite. Same for pressing the trigger button. Requiring you to fully press the trigger feels more intentful, and I've never had an accidental click on Resonite that I didn't actually mean to do, which has happened on multiple occasions on other platforms.

**Comment by shiftyscales** [comment]
As of build `2025.2.25.1316` we have a "Sticky Grab" option which should help prevent accidentally dropping things, @Narazaka.

> - Added "Sticky Grab" option to interactions
> -- When enabled, first press of grab starts holding items, second press releases
> -- This is useful when you want to hold something for longer period of time without having to press the button
> -- The toggle is in "Grabbing" submenu on each hand. The hand grab types were moved to this submenu as well to consolidate
> -- This only affects worldspace grabbing
> -- This is in VR only currently (we can enable this for desktop too if there's interest)
